leetcode/medium/227._Basic_Calculator_II.py

```diff
@@ -0,0 +1,51 @@
+# Link to problem: https://leetcode.com/problems/basic-calculator-ii/description/
+# Loom explaining solution you wrote (5m): 
+# Solution:
+# Description
+# Key ideas
+# To step process, process */ first, +- last.
+# - can be treated as + (-val), +- can be mapped as sum
+# When processing */, use a value stack
+# Difficulty: Medium
+# Code
+
+
+def token_of(s):
+    cur = ""
+    for c in s:
+        if c == " ":
+            if cur:
+                yield int(cur)
+                cur = ""
+        elif c in "+-*/":
+            if cur:
+                yield int(cur)
+            cur = ""
+            yield c
+        else:
+            assert c.isnumeric()
+            cur += c
+    if cur:
+        yield int(cur)
+
+class Solution:
+    def calculate(self, s: str) -> int:
+        itr = token_of(s)
```

Comment:
Try to think of better variable names. What is in this `itr` variable? it should be clear from the name.

Reply:
Let's pick a name that describes it better. Maybe `tokens`?